scripts/sync_data.js

```diff
@@ -170,91 +230,191 @@ async function getVideos(showMap) {
 					continue
 				}
 
-				const showId = makeIdentifier(subject)
-				if (!Object.hasOwn(showMap, showId)) {
-					showMap[showId] = {
+				const showId = toIdentifier(subject)
+				if (!Object.hasOwn(shows, showId)) {
+					shows[showId] = {
 						id: showId,
 						title: subject,
 						description: '',
 						videos: [],
 					}
 				}
 
-				showMap[showId].videos.push(videoId)
+				video.show = showId
+				shows[showId].videos.push(videoId)
 			}
 
 			videos.push(video)
 		}
 
 		if (Object.hasOwn(data, 'cursor')) {
 			params.cursor = data.cursor
```

Comment:
Mutating the `params` feels worse now that we're passing it into another function. Even though everything's done serially here so there's no danger, this mutation would be best avoided so it doesn't become a footgun if/when things get more parallel.

Also applies to `fetchGiantBombVideos` and `fetchShows`

Reply:
I agree.. it feels _off_ to do this. One thing I dislike about JS is that it's not obvious that arrays and objects are passed by reference and can be modified by the called function. In something like Rust you would specify that it's a mutable reference which would work like a warning that the function is going to modify things.

I've changed the functions to return the `shows` as well as `videos` to make it clear that they are being modified, essentially just pretending that they're being passed by value. It doesn't _fix_ the code smell, but if we just pretend that it's impossible to modify passed variables then we're less likely to introduce subtle bugs.

Reply:
A quick fix for the `params` would just be to pass in `{ ...params }` instead of the object itself.

Reply:
It's not a bad idea, so feel free to make that change if you wish... since you've approved this PR I'm just going to merge 🤠